### PR TITLE
LRP-4672 tag_releases

### PR DESCRIPTION
**Pull request by pgydev**

### CODE REVIEW

File: release/release_gold.sh [hunk 1]
```diff
@@ -5,8 +5,33 @@ source _product_info_json.sh
 source _promotion.sh
 source _publishing.sh
 
+RELEASE_PROPERTIES_FILE="release-data/release.properties"
+LIFERAY_RELEASE_FORK_NAME=${LIFERAY_RELEASE_FORK_NAME:="liferay/liferay-portal.ee"}
+
+function tag_release {
+	if ( ! curl --data "{
```
Comment (tomposmiko) [inline release/release_gold.sh]: Adjust curl switches with retry and max timeout pls.
In this case, the timeout value probably should be relatively small. We should consult with Zsolt about that.

File: release/release_gold.sh [hunk 2]
```diff
@@ -5,8 +5,33 @@ source _product_info_json.sh
...
+		--fail \
+		--header "Accept: application/vnd.github+json" \
+		--header "Authorization: Bearer ${LIFERAY_RELEASE_GITHUB_PAT}" \
+		--header "X-GitHub-Api-Version: 2022-11-28" \
+		--location https://api.github.com/repos/"${LIFERAY_RELEASE_FORK_NAME}"/git/tags \
+		--request POST \
+		--silent
+		)
+	then
+		lc_log ERROR "The requested URL could not be accessed."
```
Comment (tomposmiko) [inline release/release_gold.sh]: "Unable to access the requested URL"

File: release/release_gold.sh [hunk 3]
```diff
@@ -5,8 +5,33 @@ source _product_info_json.sh
...
+
+		return "${LIFERAY_COMMON_EXIT_CODE_BAD}"
+	fi
+}
+
 function check_usage {
-	if [ -z "${LIFERAY_RELEASE_RC_BUILD_TIMESTAMP}" ] || [ -z "${LIFERAY_RELEASE_VERSION}" ]
+	if [ -z "${LIFERAY_RELEASE_GITHUB_PAT}" ] || [ -z "${LIFERAY_RELEASE_RC_BUILD_TIMESTAMP}" ] || [ -z "${LIFERAY_RELEASE_VERSION}" ]
```
Comment (tomposmiko) [inline release/release_gold.sh]: I changed my mind. IMO we should remove this and  exit the tag_release function with SKIPPED return code if it's not defined.

File: release/release_gold.sh [hunk 4]
```diff
@@ -40,25 +65,30 @@ function main {
...
-	echo "    LIFERAY_RELEASE_RC_BUILD_TIMESTAMP: Timestamp of the build to publish"
-	echo "    LIFERAY_RELEASE_VERSION: DXP version of the release to publish"
+    echo "    LIFERAY_RELEASE_RC_BUILD_TIMESTAMP: Timestamp of the build to publish"
+    echo "    LIFERAY_RELEASE_VERSION: DXP version of the release to publish"
 	echo ""
-	echo "Example: LIFERAY_RELEASE_RC_BUILD_TIMESTAMP=1695892964 LIFERAY_RELEASE_VERSION=2023.q3.0 ${0}"
+    echo "Example: LIFERAY_RELEASE_FORK_NAME=<username/repository_name> LIFERAY_RELEASE_GITHUB_PAT=<personal access token> LIFERAY_RELEASE_RC_BUILD_TIMESTAMP=1695892964 LIFERAY_RELEASE_VERSION=2023.q3.0 ${0}"
 
-	exit "${LIFERAY_COMMON_EXIT_CODE_HELP}"
+    exit "${LIFERAY_COMMON_EXIT_CODE_HELP}"
```
Comment (tomposmiko) [inline release/release_gold.sh:98]: What was changed here?

Comment (pgydev) [reply release/release_gold.sh:98]: it got lower in lines because of the changes

File: release/release_gold.sh [hunk 5]
```diff
@@ -5,6 +5,38 @@ source _product_info_json.sh
...
+			--max-time 10 \
+			--silent
+			)
+		then
+			lc_log ERROR "Unable to access the requested URL."
+
+			return "${LIFERAY_COMMON_EXIT_CODE_BAD}"
+		fi
+	else
+		return "${LIFERAY_COMMON_EXIT_CODE_SKIPPED}"
```
Comment (tomposmiko) [inline release/release_gold.sh:36]: Indicate with message, why it is skipped.

File: release/release_gold.sh [hunk 6]
```diff
@@ -40,25 +72,30 @@ function main {
 
 	lc_time_run promote_packages
 
+	lc_time_run tag_release
+
 	promote_boms
 
 	update_product_info_json
+
```
Comment (tomposmiko) [inline release/release_gold.sh:80]: No extra line needed.

File: release/release_gold.sh [hunk 7]
```diff
@@ -40,25 +72,30 @@ function main {
...
 	echo "    LIFERAY_RELEASE_NEXUS_REPOSITORY_PASSWORD: Nexus user's password"
 	echo "    LIFERAY_RELEASE_NEXUS_REPOSITORY_USER: Nexus user with the right to upload BOM files"
 	echo "    LIFERAY_RELEASE_PRODUCT_NAME (optional): Set to \"portal\" for CE. The default is \"DXP\"."
-	echo "    LIFERAY_RELEASE_RC_BUILD_TIMESTAMP: Timestamp of the build to publish"
-	echo "    LIFERAY_RELEASE_VERSION: DXP version of the release to publish"
+    echo "    LIFERAY_RELEASE_RC_BUILD_TIMESTAMP: Timestamp of the build to publish"
+    echo "    LIFERAY_RELEASE_VERSION: DXP version of the release to publish"
 	echo ""
-	echo "Example: LIFERAY_RELEASE_RC_BUILD_TIMESTAMP=1695892964 LIFERAY_RELEASE_VERSION=2023.q3.0 ${0}"
+    echo "Example: LIFERAY_RELEASE_FORK_NAME=<username/repository_name> LIFERAY_RELEASE_GITHUB_PAT=<personal access token> LIFERAY_RELEASE_RC_BUILD_TIMESTAMP=1695892964 LIFERAY_RELEASE_VERSION=2023.q3.0 ${0}"
```
Comment (tomposmiko) [inline release/release_gold.sh:96]: Since it's optional, you can remove this change.

File: release/release_gold.sh (hunk 4 above)
Comment (tomposmiko) [reply release/release_gold.sh:98]: I don't understand, let's discuss it by slack.

File: release/release_gold.sh [hunk 8]
```diff
@@ -40,25 +72,30 @@ function main {
...
-	echo "Usage: LIFERAY_RELEASE_RC_BUILD_TIMESTAMP=<timestamp> LIFERAY_RELEASE_VERSION=<version> ${0}"
-	echo ""
-	echo "The script reads the following environment variables:"
-	echo ""
+    echo "Usage: (LIFERAY_RELEASE_FORK_NAME=<username/repository_name>) LIFERAY_RELEASE_GITHUB_PAT=<Personal Access Token> LIFERAY_RELEASE_RC_BUILD_TIMESTAMP=<timestamp> LIFERAY_RELEASE_VERSION=<version> ${0}"
+    echo ""
+    echo "The script reads the following environment variables:"
+    echo ""
+	echo "    LIFERAY_RELEASE_FORK_NAME (optional): The repository owner's git username, defaults to 'liferay' and the repository's name"
+    echo "    LIFERAY_RELEASE_GITHUB_PAT: The user's Personal Access Token"
```
Comment (tomposmiko) [inline release/release_gold.sh:89]: Indicate that it's optional.

**Pull request closed without merging**